mipidsi/src/interface/mod.rs

```diff
@@ -0,0 +1,162 @@
+//! Interface traits and implementations
+
+mod spi;
+use embedded_graphics_core::pixelcolor::{Rgb565, Rgb666};
+pub use spi::*;
+
+mod parallel;
+pub use parallel::*;
+
+/// Command and pixel interface
+pub trait Interface {
+    /// The native width of the interface
+    ///
+    /// In most cases this will be u8, except for larger parallel interfaces such as
+    /// 16 bit (currently supported)
+    /// or 9 or 18 bit (currently unsupported)
+    type Word: Copy;
+
+    /// Error type
+    type Error: core::fmt::Debug;
+
+    /// Send a command with optional parameters
+    ///
+    /// [`Interface::flush`] must be called to ensure the data is actually sent
+    fn send_command(&mut self, command: u8, args: &[u8]) -> Result<(), Self::Error>;
+
+    /// Send a sequence of pixels
+    ///
+    /// `WriteMemoryStart` must be sent before calling this function
+    ///
+    /// [`Interface::flush`] must be called to ensure the data is actually sent
+    fn send_pixels<const N: usize>(
+        &mut self,
+        pixels: impl IntoIterator<Item = [Self::Word; N]>,
+    ) -> Result<(), Self::Error>;
+
+    /// Send the same pixel value multiple times
+    ///
+    /// `WriteMemoryStart` must be sent before calling this function
+    ///
+    /// [`Interface::flush`] must be called to ensure the data is actually sent
+    fn send_repeated_pixel<const N: usize>(
+        &mut self,
+        pixel: [Self::Word; N],
+        count: u32,
+    ) -> Result<(), Self::Error>;
+
+    /// Sends any remaining buffered data
+    fn flush(&mut self) -> Result<(), Self::Error>;
```

Comment:
I'm leaning towards removing it for now too, but adding it back later is technically a breaking change.

It wouldn't break any *implementers* of the trait, but all *users* of the trait will need to add flushes. This isn't a problem if `mipidsi` is the only user because we can just add the flushes when we update the trait, but it might cause problems for anyone who is using the trait directly.

I think it's probably unlikely to cause a problem in practice, so we could just pretend that it is non-breaking.

Reply:
That's a good point, I hadn't considered that other crates might be *users* of the trait at this point in time. My suggestion would to ignore the possible breaking changes at the moment and deal with them later, if necessary.

The current solution, where the `Interface` trait is defined in the `mipidsi` crate isn't a good long term solution anyway in this regard. A breaking change in any part of `mipidsi` will make all external `Interface` impls incompatible, even if the change wasn't related to the `Interface` trait.